ERA/ERi.nls

```diff
@@ -1,43 +1,96 @@
+  let curr-soil-health 0
+  let curr-eco "A"
+  while [eco < num-ecoregions] [
+    set curr-eco (index-to-ecoregion eco)
+    ifelse count patches with [ecoregion = curr-eco] = 0 [
+      set curr-soil-health 0
+    ]  [
+      set curr-soil-health min (list 1 ((mean [soil-health] of patches with [ecoregion = curr-eco] - max-soil-health / 2) / max-soil-health / 2)) ;;normalizes worst case to -1 and best case to 1
     ]
+    table:put soil-health-table curr-eco curr-soil-health
+    table:put new-ERiE curr-eco (Bezier (curr-soil-health) 0 1)
+    set eco eco + 1
+  ]
+  set PREVIOUS-SOIL-HEALTH soil-health-table
+  set ERiE new-ERiE
+end
+
+to ERi.recalculate-ERiC ;called by ops-nodes
+  let eco 0
+  let eric-temp 0
+  while [eco < num-ecoregions] [
+    set eric-temp eric-temp + ((table:get ERiE (index-to-ecoregion eco)) * (ecoregion-percent (index-to-ecoregion eco) who))
+    set eco eco + 1
   ]
-;  show CURRENT-EXCHANGE-RATES
-;  set BASE-YEAR-EXCHANGE-RATES original-base-year-exchange-rates
+  if node-jurisdiction != "decentralized" [ set ERiC eric-temp ]
+end
+
+to-report ecoregion-percent [eco jur]
+  if count patches with [jurisdiction = jur] = 0 [report 0] ;; this is a catch all for decentralized op-node and op-nodes with no patches
+  report count patches with [Ecoregion = eco and jurisdiction = jur] / count patches with [jurisdiction = jur]
 end
 
+to-report Bezier [unadjusted-erie lower-bound upper-bound] ;https://javascript.info/bezier-curve inputs the soil health and outputs the 
+  ;numbers currently hardcoded but we can change that later with sliders
+  if unadjusted-erie >= 1 [report 0.3]
+  if unadjusted-erie <= -1 [report 10]
+  
+  let t ((upper-bound - lower-bound) / 2) + lower-bound
+  (ifelse precision ((1 - t) ^ 3 + 0.2 * (1 - t) ^ 2 * t + -3 * (1 - t) * t ^ 2 + -1 * t ^ 3) 2 = precision unadjusted-erie 2 [
+    report 0.3 * (1 - t) ^ 3 + 3 * (1 - t) ^ 2 * t + 0.15 * (1 - t) * t ^ 2 + 10 * t ^ 3
+  ]  (1 - t) ^ 3 + 0.2 * (1 - t) ^ 2 * t + -3 * (1 - t) * t ^ 2 + -1 * t ^ 3 > unadjusted-erie [ 
+    report (Bezier unadjusted-erie t upper-bound)
+  ]  [
+    report (Bezier unadjusted-erie lower-bound t)
+  ])
+    
+  
+end
+
+
+to-report index-to-ecoregion [index]
+  report (item index "ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+end
 
-to-report convert-currency [origin-currency target-currency amount-in-origin-currency] ;; currencies should be numbers 0,1,2 or "other"
+to-report convert-currency [origin-currency target-currency amount-in-origin-currency] ;; currencies should be numbers corresponding the "who" of the ops-nodes
   let origin-currency-index origin-currency
-  if origin-currency = "decentralized" or origin-currency = "other" [ set origin-currency-index 3 ]
-  let target-currency-index target-currency
-  if target-currency = "decentralized" or target-currency = "other" [ set target-currency-index 3 ]
+  let target-currency-index target-currency 
+  if origin-currency-index = "decentralized" [set origin-currency-index 3]
```

Comment:
are decentralized indices always 3? (same for next line)

Reply:
Sorry, legacy code from when num-jurisdiction was hardcoded to 3, this would have represented the decentralized currency